dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -3803,26 +3820,196 @@ Sedp::TypeLookupRequestReader::process_get_types_request(const XTypes::TypeLooku
   return DDS::RETCODE_NO_DATA;
 }
 
+void Sedp::TypeLookupRequestReader::gen_continuation_point(XTypes::OctetSeq32& cont_point) const
+{
+  // We are sending all dependencies of requested types in 1 reply, and thus
+  // the continuation_point will be "absent" by setting its length to zero.
+  cont_point.length(0);
+}
+
 DDS::ReturnCode_t
-Sedp::TypeLookupReplyReader::process_tl_reply(DCPS::Serializer& ser)
+Sedp::TypeLookupRequestReader::process_get_dependencies_request(const XTypes::TypeLookup_Request& request,
+  XTypes::TypeLookup_Reply& reply)
+{
+  // Send all dependencies of the requested types
+  sedp_.type_lookup_service_->get_type_dependencies(request.data.getTypeDependencies.type_ids,
+    reply._cxx_return.getTypeDependencies.result.dependent_typeids);
+  if (reply._cxx_return.getTypeDependencies.result.dependent_typeids.length() > 0) {
```

Comment:
Are there cases where length == 0 is expected?

Reply:
It's possible. There may be a request for TypeIdentifier that has no dependencies. I'll fix this.